### PR TITLE
project-survey

### PR DESCRIPTION
**Pull request by idanaslund**
Link to my deployed project: https://elated-sinoussi-ce6c09.netlify.app/

My reflections on how this week's project turned out: I could have done a better job on my design but I put the most time in to making all funtions in the form working. 

Things I'd like to have clarified or explained in more detail: The first day for this project went into fixing errors on the project from last week. With also working part time it is difficult to keep up with red and black level requirements.

### CODE REVIEW

File: code/public/index.html [hunk 1]
```diff
@@ -4,6 +4,15 @@
   <head>
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link href="https://fonts.googleapis.com/css2?family=Fredoka&family=Pacifico&display=swap" rel="stylesheet"/>
+
+     <!--og/meta tags for sharing on social media-->
+     <meta property="og:title" content="Ida Näslund, project-survey">
```
Comment (emmajosefina) [inline code/public/index.html:10]: Nice job with the og-tags!! Looks very professional

File: code/src/Components/FirstQ.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+
+const FirstQuestion = ({name, setName,  nextSection, handleSubmit}) => {
+    return (
+        
+            <article >
+                <div className="text-group">
+                    <h1>Welcome to the music survey</h1>
+                    <label htmlFor="name"
```
Comment (emmajosefina) [inline code/src/Components/FirstQ.js:9]: Good job! 🥇 

If you would like, you can also add:
 id="name" in the <input-tag>, then you will automatically bump into the name-field when pressing the text "Enter your name"

File: code/src/Components/FirstQ.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                        className="name-input"
+                        type="text" 
+                        name="name" 
+                        onChange={(event)=>setName(event.target.value)} 
+                        value={name} />
+                    </label> 
+                </div>
+
+                <div className="btn-group">
+                <button className="next-btn" disabled={name === ""}
```
Comment (emmajosefina) [inline code/src/Components/FirstQ.js:23]: Wow, didn't know about this disabled-thing. Great that the user can't push the button without typing his/her name in the field, and also that your button turns dark pink when writing the name in the field

File: code/src/Components/Form.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
+import React, {useState} from 'react'
+import Summary from './Summary'
+import FirstQuestion from './FirstQ'
+import SecondQuestion from './SecondQ'
+import ThirdQuestion from './ThirdQ'
+import FourthQuestion from './FourthQ'
+
+const Form = () => {
+    const [name, setName] = useState('')
```
Comment (emmajosefina) [inline code/src/Components/Form.js:9]: Looks very clean when you categorized the code like this!

File: code/src/Components/Form.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    const [amountOfBands, setAmountOfBands] = useState(false)
+    const [bands, setBands] = useState('Korn')
+    const [musicType, setMusicType] = useState([])
```
Comment (emmajosefina) [inline code/src/Components/Form.js:12]: Like the way how you're naming the variables

File: code/src/Components/Form.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    const bandOptions = ['Korn', 'Greta van Fleet', 'Red Hot Chili Peppers', 'Foo Fighters', 'Rammstein', ]
+
+    const handleSubmit = (event) => {event.preventDefault()}
+
+
+    //Removing or adding musictype in the array from checked checkboxes
+    const onCheckboxChange = (music) => {
+        if (musicType.includes(music)) {
+            const filteredMusicType = musicType.filter(item => {
+                return item !== music                     //If musictype is included it will remove it if clicked again
```
Comment (emmajosefina) [inline code/src/Components/Form.js:33]: Thanks for writing out the comment, this one was pretty hard to understand for me. Great job making it work!!

File: code/src/Components/Form.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+            setMusicType(filteredMusicType)
+        } else {
+            setMusicType([...musicType, music])
+        }
+    }
+
+return (
+    <section>
+     <div className="container">
+        <form onSubmit={handleSubmit}>
```
Comment (emmajosefina) [inline code/src/Components/Form.js:44]: Great that you added this function to the whole form, I got inspired by you!

File: code/src/Components/Form.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+            <FourthQuestion 
+            musicType={musicType}
+            onCheckboxChange={onCheckboxChange}
+            nextSection={nextSection}
+            backSection={backSection}
+            />
+            )}
+
+            {section === 4 && (
+            <Summary name = {name}
```
Comment (emmajosefina) [inline code/src/Components/Form.js:80]: I see you're writing the code with space between here, and not for the other questions. Is there a reason to that? 💯

File: code/src/Components/FourthQ.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+return (
+
+    <article>
+        <div className="text-group">
+            <h2>What type of music do you generally listen to?</h2>
+
+                <div className="checkbox-options">
+                    <label htmlFor="thegoodkind" 
+                    key="thegoodkind">
+                        <input className="checkboxes"
```
Comment (emmajosefina) [inline code/src/Components/FourthQ.js:14]: If you add id="thegoodkind" to this input you will be able to press the text to check the checkbox 💯

File: code/src/Components/SecondQ.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+const SecondQuestion = ({ setAmountOfBands, arrayOfBandNumbers, nextSection, backSection }) => {
+    
+    return (
+        <article>
+            
+                <div className="text-group">
+                    <h2>How many different bands do you listen to in one week?</h2>
+
+                    {arrayOfBandNumbers.map(amountOfBands => (
```
Comment (emmajosefina) [inline code/src/Components/SecondQ.js:11]: Really great job with mapping these! I still find it difficult to do that

File: code/src/Components/Summary.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+const Summary = ({name, amountOfBands, bands, musicType, restart}) => {
+    
+    return (
+        <article>
+            <div className="text-group">
+
+                <p>Hello <span className="answer">{name}</span>, nice to have you here!</p>
+                <p>You listen to <span className="answer">{amountOfBands}</span> number of bands per week, that's cool.</p>
+                <p>You have listed <span className="answer">{bands}</span> as your favourite band, nice choice.</p>
+                <p>You generally listen to this type of music: <span className="answer">{musicType.join(', ')}</span></p>
```
Comment (emmajosefina) [inline code/src/Components/Summary.js:12]: Great job with adding the musicType.join so the music genres displays with a , in between 🥇 Thanks for teaching me!

File: code/src/index.css [hunk 12]
```diff
@@ -1,13 +1,128 @@
...
+  min-width:200px;
+  padding: 10px;
+  margin-top: 20px;
+  border-radius: 5px;
+  border:none;
+  color:black;
+  font-size: 18px;
+}
+
+button:hover {
```
Comment (emmajosefina) [inline code/src/index.css:101]: Like this a lot!

File: code/src/index.css [hunk 13]
```diff
@@ -1,13 +1,128 @@
...
+  box-shadow: 10px 10px 26px -7px rgba(0,0,0,0.82);
+  -webkit-box-shadow: 10px 10px 26px -7px rgba(0,0,0,0.82);
+  -moz-box-shadow: 10px 10px 26px -7px rgba(0,0,0,0.82);
+  color: rgb(143, 90, 118);
+}
+
+button:disabled {
```
Comment (emmajosefina) [inline code/src/index.css:108]: 🥇

**Review by emmajosefina — APPROVED** [review]
I love this survey project Ida! You really did a great job this week, and also helped me a lot like usual 🥇 

Your survey works perfectly, and I also like that I can use my keyboard and press enter to answer the majority of the questions. You have definitely fulfilled the blue requirements for this week and also touched the red ones (in the instructions they wrote about validation attributes, such as required) but you've made it perfectly with your `<button>-tags`, not being able to hit the button without filling in the field. 

Other than tat, your survey contains:
- At least 3 questions
- One of them containing radio buttons
- One of them containing dropdown
- A submit button and a nice looking summary
- A great responsive design (I'm amazed you manage to create such a good design even though you work part time!!) That's really impressive and something you should be proud of!

The page also follows the accessibility guidelines with no errors on validator HTML checker: https://validator.w3.org/nu/?doc=https%3A%2F%2Felated-sinoussi-ce6c09.netlify.app%2F

GREAT JOB! and good luck with this week's project! 👍 🥇 💯 I'm excited to follow your developer journey!

File: code/src/Components/Form.js (hunk 8 above)
Comment (idanaslund) [reply code/src/Components/Form.js:80]: Thanks! I've made changes here and probably added spaces without thinking of it. They should not be there.

File: code/src/Components/FourthQ.js (hunk 9 above)
Comment (idanaslund) [reply code/src/Components/FourthQ.js:14]: Thank you for the tip! Will do it 👍

**Pull request closed without merging**